### PR TITLE
[codex] Implement keyboard drag-and-drop announcements

### LINKED ISSUE #161: task: Implement keyboard DnD protocol and screen reader announcements in ars-interactions
**Wave 4-3c** | **Points: 3** | **Epic: #4 (Interactions)**

Decomposed from #78.

## Goal

Implement the keyboard drag-and-drop modal protocol (Enter/Tab/Escape) and the `DragAnnouncements` screen reader integration, completing the full accessibility story for drag-and-drop.

## Layer

Subsystem

## Framework

None

## Test tier

Unit

## Depends on

- #159 (DnD core types)
- #160 (DnD state machines + use_drag/use_drop)

## Spec refs

- `spec/foundation/05-interactions.md` §7.7 "Keyboard Drag and Drop Protocol" (L2168–L2196)
- `spec/foundation/05-interactions.md` §7.8 "Screen Reader DnD Announcements" (L2198–L2268)
- `spec/foundation/04-internationalization.md` §7.1 (MessageFn pattern)

## Files to create/modify

- `crates/ars-interactions/src/drag_drop.rs` (extend with keyboard registry + announcements)

## Tests to add first

- **KeyboardDragRegistry:**
  - `new()` starts with empty targets and `current_index: None`.
  - `register(target)` appends to `targets` vec.
  - `next()` advances `current_index` (wraps around).
  - `prev()` decrements `current_index` (wraps around).
  - `current()` returns the target at `current_index`.
  - `clear()` resets targets and index to initial state.
  - Tab cycling with 3 targets: cycles 0 → 1 → 2 → 0.
  - Shift+Tab cycling: cycles 2 → 1 → 0 → 2.
  - Single target: Tab stays on same target.
- **DragAnnouncements:**
  - `default()` produces non-empty strings for all fields.
  - `drag_start` with 1 item contains "1 item".
  - `drag_start` with 3 items contains "3 items".
  - `drag_enter` includes the target name.
  - `drop` includes both item count and target name.
  - `drag_cancel` produces "Drag cancelled".
- **Keyboard protocol integration:**
  - Enter on focused draggable transitions source to Dragging.
  - Tab in keyboard-drag mode calls `registry.next()` and fires `on_drag_enter`.
  - Enter on focused drop target fires `on_drop` and transitions to Dropped.
  - Escape cancels drag, fires `on_drag_end(was_dropped: false)`, clears registry.

## Acceptance criteria

- `KeyboardDragRegistry` struct with `targets: Vec<KeyboardDropTarget>`, `current_index: Option<usize>`.
- `KeyboardDropTarget` struct with `element_id: String`, `label: String`, `config: DropConfig`.
- Registry supports: register, unregister, next, prev, current, clear.
- Full keyboard protocol: Enter → start, Tab/Shift+Tab → cycle targets, Enter → drop, Escape → cancel.
- `DragAnnouncements` struct with 5 `MessageFn` fields per spec §7.8.
- `Default` impl produces English announcement strings.
- `Debug` impl uses `finish_non_exhaustive()`.
- Announcement dispatch uses `AnnouncementPriority::Assertive` for drag_start/drop/drag_cancel and `Polite` for drag_enter/drag_leave per spec.
- Per-element announcement closures on DragConfig/DropConfig take precedence over DragAnnouncements defaults.
- All `MessageFn` closures carry `Send + Sync` bounds per project convention.
- All public items have `///` doc comments.
- Zero warnings under workspace lint config.

## Spec impact

No spec change required.

### PR DESCRIPTION
**Pull request by fogodev**
## Summary
- implement keyboard drag-and-drop announcements and registry support in `ars-interactions`
- add the `ars-interactions -> ars-a11y` dependency and the missing `MessageFn` arities used by DnD messaging
- sync the foundation spec text for the new dependency, keyboard helper surface, and explicit keyboard target registration model

## Validation
- `cargo test -p ars-interactions drag_drop --lib`
- `cargo llvm-cov test -p ars-interactions --text --lib drag_drop`
- `cargo xci`

Closes #161

### CODE REVIEW

File: crates/ars-interactions/src/drag_drop.rs [hunk 1]
```diff
@@ -22,6 +23,170 @@ type DragItemsFn = Arc<dyn Fn() -> Vec<DragItem> + Send + Sync>;
...
+
+    /// Returns the currently selected target, if any.
+    #[must_use]
+    pub fn current(&self) -> Option<&KeyboardDropTarget> {
+        self.current_index.and_then(|index| self.targets.get(index))
+    }
+
+    /// Clears the registry and active selection.
+    pub fn clear(&mut self) {
+        self.targets.clear();
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-interactions/src/drag_drop.rs]: Fixed in `e96f77d`.

`KeyboardDragRegistry::clear()` now resets only `current_index` and preserves registered targets, so ending a keyboard drag no longer discards mounted targets. I also added/updated tests to verify targets remain reachable after both successful drop and cancel, plus the no-op edge paths.

Validation: `cargo test -p ars-interactions drag_drop --lib` and `cargo xci`.

File: crates/ars-interactions/src/drag_drop.rs [hunk 2]
```diff
@@ -22,6 +23,170 @@ type DragItemsFn = Arc<dyn Fn() -> Vec<DragItem> + Send + Sync>;
...
+        Self::default()
+    }
+
+    /// Registers a live drop target in document order.
+    pub fn register(&mut self, target: KeyboardDropTarget) {
+        self.targets.push(target);
+
+        if self.current_index.is_none() {
+            self.current_index = Some(0);
+        }
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-interactions/src/drag_drop.rs]: Fixed in `e96f77d`.

`register()` no longer preselects index `0`, so the registry starts with no active selection and the first `next()` now lands on the first target in document order. The registry tests were updated to cover the first-step navigation behavior explicitly.

Validation: `cargo test -p ars-interactions drag_drop --lib` and `cargo xci`.

File: crates/ars-interactions/src/drag_drop.rs [hunk 3]
```diff
@@ -554,6 +747,27 @@ impl DragResult {
...
+        registry: &mut KeyboardDragRegistry,
+        locale: &Locale,
+        announcer: &mut LiveAnnouncer,
+        announcements: &DragAnnouncements,
+    ) -> Option<DragEndEvent> {
+        let event = self.cancel_drag()?;
+
+        registry.clear();
+
+        announcer.announce_with_priority(
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-interactions/src/drag_drop.rs:765]: Fixed in `3d1e30e`.

`cancel_keyboard_drag(...)` now takes the active `DropResult` and resets it as part of keyboard cancel, so stale `drag_over` / drop-indicator state is cleared along with the source state and registry selection. I also updated the cancel-path tests to assert the target snapshot is cleared.

Validation: `cargo test -p ars-interactions drag_drop --lib` and `cargo xci`.

File: crates/ars-interactions/src/drag_drop.rs [hunk 4]
```diff
@@ -22,6 +23,165 @@ type DragItemsFn = Arc<dyn Fn() -> Vec<DragItem> + Send + Sync>;
...
+
+        self.targets.remove(index);
+
+        self.current_index = if self.targets.is_empty() {
+            None
+        } else {
+            self.current_index
+                .map(|current| current.saturating_sub(usize::from(index < current)))
+                .map(|current| current.min(self.targets.len() - 1))
+        };
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-interactions/src/drag_drop.rs]: Fixed in `3d1e30e`.

`KeyboardDragRegistry::unregister()` now clears selection when the currently selected target is removed instead of implicitly retargeting to a different entry. That keeps registry selection aligned with explicit keyboard navigation and announcements. I added a focused test for the selected-target removal case.

Validation: `cargo test -p ars-interactions drag_drop --lib` and `cargo xci`.
